### DISCUSSION TITLE
Graceful way to fallback to default retry

### DISCUSSION THREAD

**Discussion by flex-mookeun** [TanStack/query#5973] [Q&A] [answered]
Hi there,

I've tried to reuse default `retry` using `getDefualtOptions`,
But I end up having same code with [the retryer inside the query](https://github.com/TanStack/query/blob/main/packages/query-core/src/retryer.ts#L167-L170).
Any idea on more graceful way of handling this?

```ts
retry: (failureCount, error) => {
  if (isMySpecialErrorCase(error)) {
    return false;
  }

  const defaultRetry = queryClient.getDefaultOptions().queries?.retry;
  
  // same with retryer inside the query
  if (typeof defaultRetry === 'function') {
    return defaultRetry(failureCount, error);
  }
  if (typeof defaultRetry === 'number') {
    return failureCount < defaultRetry;
  } 
  return defaultRetry ?? false;
}
```

**Comment by TkDodo** [comment]
default retry is just the number 3 🤷

**Reply by flex-mookeun** [reply]
I'm using my custom retry function to ignore auth errors and passing it to `QueryClient.defaultOptions.queries.retry`. 
I know it will be a function on runtime, I wanted to make it type safe.

**Reply by TkDodo** [reply, marked as answer]
don't think you can.
